src/consumer/base_consumer.rs

```diff
@@ -313,6 +313,25 @@ impl<C: ConsumerContext> Consumer<C> for BaseConsumer<C> {
         }
     }
 
+    fn offsets_for_times<T: Into<Option<Duration>>>(&self, mut timestamps: TopicPartitionList, timeout: T)
+                                                    -> KafkaResult<TopicPartitionList>
+    {
+        // This call will then put the offset in the offset field of this topic partition list.
+        let offsets_for_times_error = unsafe {
```

Comment:
This code seems to be the same code used in `offsets_for_timestamp`. Would it make sense to refactor `offsets_for_timestamp` to call this new function?